File: backend/src/main/java/br/ufpr/sept/so2/modules/solicitacoes/infrastructure/persistence/ProtocoloSeqJpaEntity.java
package br.ufpr.sept.so2.modules.solicitacoes.infrastructure.persistence;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "solicitacao_protocolo_seq")
public class ProtocoloSeqJpaEntity {

    @Id
    private Integer ano;

    @Column(nullable = false)
    private int ultimo;

    protected ProtocoloSeqJpaEntity() {
    }

    public ProtocoloSeqJpaEntity(Integer ano, int ultimo) {
        this.ano = ano;
        this.ultimo = ultimo;
    }

    public Integer getAno() {
        return ano;
    }

    public int getUltimo() {
        return ultimo;
    }

    public void setUltimo(int ultimo) {
        this.ultimo = ultimo;
    }
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/solicitacoes/infrastructure/persistence/ProtocoloSeqJpaEntity.kt
package br.ufpr.sept.so2.modules.solicitacoes.infrastructure.persistence

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "solicitacao_protocolo_seq")
class ProtocoloSeqJpaEntity {

    @field:Id
    var ano: Int? = null

    @field:Column(nullable = false)
    var ultimo: Int = 0

    protected constructor()

    constructor(ano: Int?, ultimo: Int) {
        this.ano = ano
        this.ultimo = ultimo
    }
}
